firmware/library/L1_Drivers/timer.hpp

```diff
@@ -0,0 +1,157 @@
+    kRegCR0 = 4,
+    kRegCR1 = 5
+  };
+  enum TimerPort : uint8_t
+  {
+    kTimer0 = 0,
+    kTimer1 = 1,
+    kTimer2 = 2,
+    kTimer3 = 3
+  };
+  // Modes for each match register that when match register matches timer
+  // system can trigger an Interrupt, Stop Counter, or Restart and in any
+  // mixture of those commands.
+  enum TimerSetting : uint8_t
+  {
+    kInterrupt            = 0b001,
+    kRestart              = 0b010,
+    kStop                 = 0b100,
+    kInterruptRestart     = 0b011,
+    kInterruptStop        = 0b101,
+    kRestartStop          = 0b110,
+    kInterruptRestartStop = 0b111
+  };
+  virtual void Initialize(TimerPort timer)= 0;
+  virtual void SetTimer(uint32_t time, RegisterChoice reg,
+                        TimerSetting mode) = 0;
+  virtual uint32_t GetTimer() = 0;
+  virtual void ClearInterrupts() = 0;
+};
+
+class Timer : public TimerInterface, protected Lpc40xxSystemController
+{
+ public:
+  static constexpr uint8_t kTime0Mat0   = 0b011;
+  static constexpr uint8_t kClearMode   = 0b0111;
+  static constexpr uint8_t kPrescaleMicro   = 48;
+  static constexpr uint32_t kClear = 0b11111111111111111111111111111111;
+
+  enum TimerInterruptBits : uint8_t
+  {
+    kMat0 = 0,
+    kMat1 = 3,
+    kMat2 = 6,
+    kMat3 = 9
+  };
+
+  inline static Pin quads[4][4] = {
+                  { Pin(1, 28), Pin(1, 29), Pin(0, 15), Pin(0, 15) },
+                  { Pin(1, 22), Pin(1, 25), Pin(0, 3),  Pin(0, 3) },
+                  { Pin(2, 5) , Pin(2, 4),  Pin(2, 3),  Pin(2, 2) },
+                  { Pin(2, 26), Pin(2, 27), Pin(2, 30), Pin(2, 31) } };
+//                  MAT 0     , Mat 1     , Mat 2     , Mat 3
+
+  explicit constexpr Timer(TimerPort mode)
+                      : channel_(static_cast<uint8_t>(mode)),
+                        mat0_(&quads[channel_][0]),
+                        mat1_(&quads[channel_][1]),
+                        mat2_(&quads[channel_][2]),
+                        mat3_(&quads[channel_][3])
+  {
+  }
+
+  void Initialize(TimerPort timer) override
+  {
+    PowerUpPeripheral(Lpc40xxSystemController::Peripherals::kTimer0);
+    PowerUpPeripheral(Lpc40xxSystemController::Peripherals::kTimer1);
+    PowerUpPeripheral(Lpc40xxSystemController::Peripherals::kTimer2);
+    PowerUpPeripheral(Lpc40xxSystemController::Peripherals::kTimer3);
+    mat0_->SetPinFunction(kTime0Mat0);
+    mat0_->SetAsAnalogMode(true);
+    mat0_->SetMode(PinInterface::Mode::kInactive);
+    mat1_->SetPinFunction(kTime0Mat0);
+    mat1_->SetAsAnalogMode(true);
+    mat1_->SetMode(PinInterface::Mode::kInactive);
+    mat2_->SetPinFunction(kTime0Mat0);
+    mat2_->SetAsAnalogMode(true);
+    mat2_->SetMode(PinInterface::Mode::kInactive);
+    mat3_->SetPinFunction(kTime0Mat0);
+    mat3_->SetAsAnalogMode(true);
+    mat3_->SetMode(PinInterface::Mode::kInactive);
+    // set Prescale register for Prescale Counter to milliseconds
+    tim_register[static_cast<uint8_t>(timer)]->PR &= ~(kClear << 1);
```

Comment:
Be sure to include `util/enum.hpp`
```suggestion
   uint8_t timer_index = util::Value(timer);
    tim_register[]->PR &= ~(kClear << 1);
```

Reply:
Do this for all other instances of static_cast of enum classes.